lib/Twig/Filter.php

```diff
@@ -34,6 +34,10 @@ class Twig_Filter
      */
     public function __construct(string $name, $callable = null, array $options = array())
     {
+        if (__CLASS__ !== get_class($this)) {
+            @trigger_error('Overriding '.__CLASS__.' is deprecated since version 2.4.0 and the class will be final in 3.0.', E_USER_DEPRECATED);
```

Comment:
wouldn't this make it painful to use Twig_SimpleFilter for now, to keep support for Twig 1.x and 2.x in parallel ? I suggest whitelisting Twig_SimpleFilter for now here.
In the past, we agreed on delaying the deprecation of Twig_SimpleFilter until soon before 3.x to make it easier for external libraries not able to drop support for Twig 1.x (for instance Symfony 3.x due to the min PHP version)